redash/models/__init__.py

```diff
@@ -204,16 +218,33 @@ def get_by_id(cls, _id):
         return cls.query.filter(cls.id == _id).one()
 
     @classmethod
-    def save_schema(self, schema_info):
+    def save_schema(cls, schema_info):
+        # There was a change in column data.
         if 'columnId' in schema_info:
             ColumnMetadata.query.filter(
-                ColumnMetadata.id == schema_info['columnId'],
                 ColumnMetadata.table_id == schema_info['tableId'],
+                ColumnMetadata.id == schema_info['columnId']
             ).update(schema_info['schema'])
-        else:
-            TableMetadata.query.filter(
+            db.session.commit()
+            return
+
+        if 'sample_queries' in schema_info['schema']:
+            sample_queries = schema_info['schema']['sample_queries']
+            del schema_info['schema']['sample_queries']
+
+            table_metadata_object = TableMetadata.query.filter(
                 TableMetadata.id == schema_info['tableId']
-            ).update(schema_info['schema'])
+            ).first()
+            table_metadata_object.sample_queries = []
+
+            for sample_query in sample_queries.values():
+                query_object = Query.query.filter(Query.id == sample_query['id']).first()
+                table_metadata_object.sample_queries.append(query_object)
```

Comment:
There is a risk of an increasing number of db queries where there are many sample queries (e.g. think TOP 3 data sources in use). Can we do this in one query separately instead of in a loop?

Reply:
Done

Reply:
👍